src/charm.py

```diff
@@ -277,6 +277,8 @@ def _on_update_status(self, _):
         # if unit is primary then update status
         if self._primary == self.unit.name:
             self.unit.status = ActiveStatus("Replica set primary")
+        else:
+            self.unit.status = ActiveStatus("")
```

Comment:
nit, if no message is set, just:
```suggestion
            self.unit.status = ActiveStatus()
```